File: backend/schemas/sync_jobs.py
from __future__ import annotations

from .common import FlexibleSchema


class SyncJobRequest(FlexibleSchema):
    id: str | None = None
    name: str | None = None
    enabled: bool | None = False
    sourceType: str | None = "local"
    syncMode: str | None = "add"
    intervalMinutes: int | None = 0
    scheduleTime: str | None = ""
    profileId: str | None = None
    sourcePath: str | None = ""
    sourceCid: str | None = "0"
    cookie: str | None = ""
    pan115AccountId: str | None = ""
    userAgent: str | None = ""
    targetDir: str | None = ""
    recursive: bool | None = True
    dedupeScope: str | None = "global"

File: frontend/apps/web-antd/src/api/sync-jobs.ts
import { del, post } from './request';
import type { SyncJob } from './dashboard';

export interface SyncJobSaveBody {
  id?: string;
  name?: string;
  enabled?: boolean;
  sourceType?: string;
  syncMode?: string;
  intervalMinutes?: number;
  scheduleTime?: string;
  profileId?: string;
  sourcePath?: string;
  sourceCid?: string;
  cookie?: string;
  pan115AccountId?: string;
  userAgent?: string;
  targetDir?: string;
  recursive?: boolean;
  dedupeScope?: string;
}

export const syncJobsApi = {
  save: (body: SyncJobSaveBody) => post<{ ok: true; job: SyncJob }>('/api/sync-jobs', body),
  run: (id: string) => post(`/api/sync-jobs/${id}/run`, {}),
  remove: (id: string) => del(`/api/sync-jobs/${id}`)
};
